pkg/api/validate/plugin.go

```diff
@@ -250,7 +250,7 @@ func validateImageConfig(path string, i *pluginapi.ImageConfig) (errs []error) {
 		errs = append(errs, fmt.Errorf("invalid %s.azureControllers %q", path, i.AzureControllers))
 	}
 
-	if i.Canary == "" {
+	if (i.Canary == "") != (version == "v3.2") {
```

Comment:
this seems correct, but is confusing. why not:
```
if (i.Canary == "") && (version != "v3.2") {
```

Reply:
because it's not equivalent? ;-)

it'd have to be `if (i.Canary == "" && version != "v3.2") || (i.Canary != "" && version == "v3.2")` ...